src/implementation/allocator.jl

```diff
@@ -148,3 +159,70 @@ end
 function tensorfree!(C, allocator=DefaultAllocator())
     return nothing
 end
+
+# ------------------------------------------------------------------------------------------
+# ManualAllocator implementation
+# ------------------------------------------------------------------------------------------
```

Comment:
Is there a reason for or benefit in overloading `tensoralloc_add` and `tensoralloc_contract` over just `tensoralloc`?

Also, I would restrict the `ManualAllocator` method to `AbstractArray`s, so maybe something like
```julia
function tensoralloc(ttype::Type{<:AbstractArray}, structure, istemp::Val{T}, ::ManualAllocator)
    if T
         return malloc(TC, structure...)
     else
         return tensoralloc(ttype, structure, istemp, DefaultAllocator())
     end
end
```
I am probably overlooking something?

Reply:
The current interface does not pass the istemp argument to the tensortype function. This means that I cannot make that type PtrArray, and it is then a bit strange to make tensoralloc return a different type from the one it got requested by its arguments. (The current implementation also has a type assert there)